### PR TITLE
Api automation 1.0

### PR DESCRIPTION
**Pull request by RafaDdS**
Creates an automation system to implement the bindings based on an input file. 
Most of the Core and ImgProc have been re-implemented using the system.
Some of the tests for the new methods.
Updated to Godot 4.3 to use in editor documentation.

### CODE REVIEW

File: SConstruct
Comment (matt-s-clark) [inline SConstruct:1]: To compile on MacOS, line 18 of SConstruct needs to be commented out. I wonder if there is an easy way to include/remove based on arch?

File: demo/Examples/CascadeClassifierExample.gd
Comment (matt-s-clark) [inline demo/Examples/CascadeClassifierExample.gd:1]: This used to recognise my fist, but I had trouble getting the classifer to draw a square on my fist this time around. The square (overlay) did pop up in a couple of places, but not on my fist.

**Review by matt-s-clark — COMMENTED** [review]
For MacOS, GUT did crash when attempting to run all of the tests. 

Test that crashes GUT:
* test_video_capture.gd

Test that failed with "Out of bounds get index'0' (on base: 'Array'):
* test_face.gd

Test that failed with "Invalid get index 'x' (on base 'Nil')"
* test_trackers.gd

Test that passed 3 items:
* test_video_capture.gd

I don't think there are any show stoppers there. Great work!

**Review by matt-s-clark — APPROVED** [review]

File: SConstruct
Comment (RafaDdS) [reply SConstruct:1]: I could set it to be included based on arch, but it's not needed on Linux. if it's also not important to Windows we could just remove it.

**Comment by RafaDdS** [comment]
I added checks to give better test outputs and made a small change that might solve the crash.

**Pull request merged**